packages/legend-extension-dsl-data-space/src/graph-manager/protocol/pure/v1/V1_DSL_DataSpace_PureGraphManagerExtension.ts

```diff
@@ -355,55 +510,140 @@ export class V1_DSL_DataSpace_PureGraphManagerExtension extends DSL_DataSpace_Pu
-      .concat(
-        uniq(
-          analysisResult.model.elements.flatMap((element) => {
-            if (element instanceof V1_Class) {
-              return element.derivedProperties
-                .map((prop) => prop.returnType)
-                .concat(element.properties.map((prop) => prop.type));
-            }
-            return [];
-          }),
-        )
-          // make sure to not include types already returned by the analysis
-          .filter(
-            (path) =>
-              !analysisResult.model.elements
-                .map((el) => el.path)
-                .includes(path),
-          )
-          .map((path) => {
-            const [pkgPath, name] = resolvePackagePathAndElementName(path);
-            if (!pkgPath) {
-              // exclude package-less elements (i.e. primitive types)
-              return undefined;
-            }
-            const _class = new V1_Class();
-            _class.name = name;
-            _class.package = pkgPath;
-            return _class;
-          })
-          .filter(isNonNullable),
-      )
-      .concat(mappingModels)
-      .concat(runtimeModels)
-      // NOTE: if an element could be found in the graph already it means it comes from system
-      // so we could rid of it
-      .filter((el) => !graph.getNullableElement(el.path, false))
-      .map((el) => this.graphManager.elementProtocolToEntity(el));
-
-    await this.graphManager.buildGraph(
-      graph,
-      graphEntities,
-      ActionState.create(),
+    // The DataSpace entity is excluded from AnalyticsResult.Json to reduce the JSON size
+    // because all its information can be found in V1_DataSpaceAnalysisResult.
+    // Therefore, we are building a simple v1_DataSpace entity based on V1_DataSpaceAnalysisResult.
+    const dataspaceEntity = new V1_DataSpace();
+    dataspaceEntity.name = analysisResult.name;
+    dataspaceEntity.package = analysisResult.package;
+    dataspaceEntity.supportInfo = analysisResult.supportInfo;
+    dataspaceEntity.executionContexts = analysisResult.executionContexts.map(
+      (execContext) => {
+        const contextProtocol = new V1_DataSpaceExecutionContext();
+        contextProtocol.name = execContext.name;
+        contextProtocol.title = execContext.title;
+        contextProtocol.description = execContext.description;
+        contextProtocol.mapping = new V1_PackageableElementPointer(
+          PackageableElementPointerType.MAPPING,
+          execContext.mapping,
+        );
+        contextProtocol.defaultRuntime = new V1_PackageableElementPointer(
+          PackageableElementPointerType.RUNTIME,
+          execContext.defaultRuntime,
+        );
+        return contextProtocol;
+      },
     );
+    dataspaceEntity.defaultExecutionContext =
+      analysisResult.defaultExecutionContext;
+    dataspaceEntity.title = analysisResult.title;
+    dataspaceEntity.description = analysisResult.description;
+
+    const resolvedMappingPath =
```

Comment:
for this, if no executionContext provided should be use the default to find the mapping ?